src/nvim/api/win_config.c

```diff
@@ -330,7 +354,90 @@ static bool parse_float_bufpos(Array bufpos, lpos_T *out)
   return true;
 }
 
-static void parse_border_style(Object style, FloatConfig *fconfig, Error *err)
+static void parse_border_title(Object title, Object title_pos, Object width, FloatConfig *fconfig,
+                               Error *err)
+{
+  if (!parse_title_pos(title_pos, fconfig, err)) {
+    return;
+  }
+
+  if (title.type == kObjectTypeString) {
+    if (title.data.string.size == 0) {
+      fconfig->title = false;
+      return;
+    }
+    fconfig->title_text = xstrdup(title.data.string.data);
+    int hl_id = object_to_hl_id(title, "border title highlight", err);
+    fconfig->title_hi_id = hl_id;
```

Comment:
as a simplification of the data structure, We could manually allocate a VirtText of size one and put a single chunk with the title_text + hl_id of "FloatBorderTitle", then we don't need to deal with this case outside of this function.

Reply:
done. refactored.